components/email/src/Email.svelte

```diff
@@ -485,19 +500,21 @@
       if (msgIndex === activeThread.messages.length - 1) {
         doNothing(event);
       } else {
-        activeThread.messages[msgIndex].expanded =
-          !activeThread.messages[msgIndex].expanded;
+        activeThread.messages[msgIndex].expanded = !activeThread.messages[
+          msgIndex
+        ].expanded;
       }
     }
   }
 
-  function fetchIndividualMessage(msgIndex: number) {
-    const messageID = activeThread.messages[msgIndex].id;
-
+  function fetchIndividualMessage(
+    msgIndex: number,
+    messageID: string,
+  ): Promise<string | null> {
     messageLoadStatus[msgIndex] = "loading";
-    fetchMessage(query, messageID).then((json) => {
-      activeThread.messages[msgIndex].body = json.body;
```

Comment:
was causing infinite loop when adding `fetchIndividualMessage` to the main "check local or fetch thread" reactive statement.

So refactored to remove assignment of `activeThread`